runtime/protos/runtime.proto

```diff
@@ -324,195 +415,30 @@ message ExportListenerReply {
-      // of keys accessed in the slice. For example, if a slice [0, 1000)
-      // receives requests for keys 0, 1, 2, 3, and 10, then the size of the
-      // slice is 5.
-      //
-      // TODO(mwhittaker): Implement this. For now, we do report the size of a
-      // slice [a, b) as b - a.
-      uint64 size = 5;
-    }
-
-    // TODO(mwhittaker): In full generality, load is a generic
-    // application-specific double-valued measurement (e.g., total number of
-    // requests, size of components in memory). For now to simplify things, we
-    // assume load is measured as requests per second.
-    repeated SliceLoad load = 1;  // load, by slice
-
-    // TODO(mwhittaker): For now, load is reported with respect to a given
-    // assignment. When the assignment changes, the load is reset. This approach
-    // works for rate-based load like requests per second, but not other
-    // non-rate based load like size of cached components.
-    uint64 version = 3;
-  }
-
-  string app = 1;                        // Application name
-  string deployment_id = 2;              // Deployment id.
-  string replica = 4;                    // e.g., tcp://124.123.123.412:1291
-  map<string, ComponentLoad> loads = 5;  // load for weavelet's components
-}
-
-// MetricType specifies a particular kind of metric.
-enum MetricType {
-  INVALID = 0;
-  COUNTER = 1;
-  GAUGE = 2;
-  HISTOGRAM = 3;
-}
-
-// MetricUpdate contains the metrics' updated values.
-message MetricUpdate {
-  // Metric definitions and updated values.
-  //
-  // All metrics are identified by their globally unique id. Two endpoints
-  // exchanging metric updates must ensure that a received metric value has been
-  // preceded by a metric definition with the same id, either in the current
-  // update or an earlier one.
-  repeated MetricDef defs = 1;      // metric definitions
-  repeated MetricValue values = 2;  // metric values
-}
-
-// MetricDef is the definition of a new metric.
-message MetricDef {
-  uint64 id = 1;
-  string name = 2;
-  MetricType typ = 3;
-  string help = 4;
-  map<string, string> labels = 5;
-  repeated double bounds = 6;  // bucket bounds, for histograms
-}
-
-// MetricValue is the value associated with a metric.
-message MetricValue {
-  uint64 id = 1;               // metric's unique id.
-  double value = 2;            // value for counter and gauge, sum for histogram
-  repeated uint64 counts = 3;  // histogram counts
-}
-
-// MetricSnapshot is a snapshot of a metric. It is the union of a MetricDef and
-// a MetricValue.
-message MetricSnapshot {
-  uint64 id = 1;
-  string name = 2;
-  MetricType typ = 3;
-  string help = 4;
-  map<string, string> labels = 5;
-  repeated double bounds = 6;
-  double value = 7;
-  repeated uint64 counts = 8;
-}
-
 // LogEntry is a log entry. Every log entry consists of a message (the thing the
 // user logged) and a set of metadata describing the message.
 message LogEntry {
```

Comment:
Actually, logs and traces aren't sent over as RPCs. Is there a reason to make UpdateRoutingInfo and UpdateComponents into RPCs then?

This is a subtle point, not sure if we want to RPC-ize everything anyway. WDYT?